libraries/aws_elasticache_replication_groups.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+require 'aws_backend'
+
+class AwsElastiCacheReplicationGroups < AwsResourceBase
+  name 'aws_elasticache_replication_groups'
+  desc 'Verifies settings for a collection of AWS Replication Groups'
+  example "
+    describe aws_elasticache_replication_groups do
+      its('count') { should eq 3 }
+    end
+
+    # Iterate through all clusters
+    aws_elasticache_replication_groups.ids.each do |replication_group_id|
+      describe aws_elasticache_replication_group(replication_group_id) do
+        it { should be_encrypted_at_rest }
+        its('engine') { should cmp 'redis' }
+      end
+    end
+  "
+
+  attr_reader :table
+
+  FilterTable.create
+             .register_column(:ids, field: :id)
+             .register_column(:node_types, field: :node_type)
+             .register_column(:status, field: :status)
+             .register_column(:encrypted_at_rest, field: :encrypted_at_rest)
+             .register_column(:encrypted_at_transit, field: :encrypted_at_transit)
+             .install_filter_methods_on_resource(self, :table)
+
+  def initialize(opts = {})
+    super(opts)
+    validate_parameters
+    @table = fetch_data
+  end
+
+  def fetch_data
+    elasticache_replication_group_rows = []
+    catch_aws_errors do
+      api_response = @aws.elasti_cache_client.describe_replication_groups
```

Comment:
this api supports pagination request 
https://docs.aws.amazon.com/sdk-for-ruby/v3/api/Aws/ElastiCache/Client.html#describe_replication_groups-instance_method
you will have to implement it for completing the functionality

Reply:
updated